class-tumblr-import.php

```diff
@@ -34,8 +59,10 @@ public function start() {
 				$this->error = null;
 			}
 
-			@$this->consumerkey = defined( 'TUMBLR_CONSUMER_KEY' ) ? TUMBLR_CONSUMER_KEY : ( ! empty( $_POST['consumerkey'] ) ? $_POST['consumerkey'] : $this->consumerkey );
-			@$this->secretkey   = defined( 'TUMBLR_SECRET_KEY' ) ? TUMBLR_SECRET_KEY : ( ! empty( $_POST['secretkey'] ) ? $_POST['secretkey'] : $this->secretkey );
+			do_action( 'tumblr_importer_import_start' );
+
+			@$this->consumerkey = defined( 'TUMBLR_CONSUMER_KEY' ) ? TUMBLR_CONSUMER_KEY : ( ! empty( $_POST['consumerkey'] ) ? sanitize_text_field( wp_unslash( $_POST['consumerkey'] ) ) : $this->consumerkey );
```

Comment:
wondering how much trouble would be to avoid using `@`  to suppress errors? also side question, what is the minimum PHP versions supported by this plugin? I didn't find it easily

Reply:
I've removed the suppression here for now and I'm not seeing any issues with the line.